browser/brave_wallet/keyring_controller_unittest.cc

```diff
@@ -1593,4 +1593,100 @@ TEST_F(KeyringControllerUnitTest, NotifyUserInteraction) {
   ASSERT_TRUE(controller.IsLocked());
 }
 
+TEST_F(KeyringControllerUnitTest, SetSelectedAccount) {
+  KeyringController controller(GetPrefs());
+  controller.CreateWallet("brave", base::DoNothing::Once<const std::string&>());
+  base::RunLoop().RunUntilIdle();
+  std::string first_account = controller.default_keyring_->GetAddress(0);
+  controller.AddAccountForDefaultKeyring("Who does number 2 work for");
+  std::string second_account = controller.default_keyring_->GetAddress(1);
+
+  // This does not depend on being locked
+  controller.Lock();
+  EXPECT_TRUE(controller.IsLocked());
+
+  // No account set as the default
+  controller.GetSelectedAccount(base::BindLambdaForTesting(
+      [&](const absl::optional<std::string>& account) {
+        EXPECT_EQ(absl::nullopt, account);
+      }));
+  base::RunLoop().RunUntilIdle();
```

Comment:
capture RunLoop and do Run and Quit if you don't check callback is called or not here.
same rule applies to all callbacks in this case

Reply:
Fixed and rebased in 790de376def0d43e16b8421e0fd9475b81ec4bd4